### PR TITLE
[VQA] User Profiles

### PR DESCRIPTION
**Pull request by pbortnick**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-pbvqa-user-profiles-hashicorp.vercel.app/profile/bookmarks) 🔎
- [Asana task](https://app.asana.com/0/1202097197789424/1202911555850623/f) 🎟️

## 🗒️ What

Update all VQA Items for User Profiles:
- Add bookmark icon next to sidebar menu item
- Make empty state cta button 'secondary' styles
- Adjust spacing between subheading and content
- Two HRs on bottom of mobile nav
- user profile mobile menu breakpoint adjustment
- Adjust toast animation duration to 5 seconds
- Account settings button opens in new tab from:
1. profile/bookmarks sidebar on desktop
2. mobile menu
3. desktop nav profile dropdown

## 📸 Design Screenshots

🎨 [Bookmark Icon Design](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=12138%3A127457)
🎨 [Empty State Button Design](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=12357%3A148760)
🎨 [Adjust spacing design](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=12357%3A149785)

## 🧪 Testing

Make sure all items listed above are addressed.
**Please test the sidebar in different areas of the site on both mobile and desktop for the [double HR issue](https://github.com/hashicorp/dev-portal/pull/1137#discussion_r984708266). I might have missed some areas that could have been affected by this update**

### CODE REVIEW

File: src/components/bookmark-button/toast/make-bookmark-toast.tsx [hunk 1]
```diff
@@ -47,6 +47,6 @@ export default function makeBookmarkToast(
 ) {
 	toast({
 		...generateToastMessage(action, tutorialName),
-		autoDismiss: 15000,
+		autoDismiss: 5000,
```
Comment (pbortnick) [inline src/components/bookmark-button/toast/make-bookmark-toast.tsx:50]: subtask: [Adjust toast animation duration to 5 seconds](https://app.asana.com/0/0/1203075873125169/f)

File: src/components/dev-dot-content/mdx-components/mdx-heading-permalink/mdx-heading-permalink.module.css [hunk 2]
```diff
@@ -28,7 +28,7 @@
 		opacity: 1;
 		background: var(--token-color-surface-faint);
 
-		& svg {
+		& .icon {
```
Comment (pbortnick) [inline src/components/dev-dot-content/mdx-components/mdx-heading-permalink/mdx-heading-permalink.module.css:31]: Unrelated to current PR (from my most previous) but threw it in here to better follow our [CSS Modules best practices](https://www.notion.so/hashicorp/CSS-Modules-753f2647f7aa4a598e28990d9f0d05a8#de383178becc4320b9e358f0982a6ecc)

File: src/lib/generate-top-level-sub-nav-items.ts [hunk 3]
```diff
@@ -43,7 +43,6 @@ export const generateTopLevelSubNavItems = () => {
 		{ divider: true },
 		{ heading: 'Products' },
 		...betaProductItems,
-		{ divider: true },
```
Comment (pbortnick) [inline src/lib/generate-top-level-sub-nav-items.ts:46]: Subtask: [[bug] Two HRs on bottom of mobile nav](https://app.asana.com/0/1202097197789424/1203075873125158/f)

Could use extra testing please!

File: src/views/profile/bookmarks-view/bookmarks-view.module.css [hunk 4]
```diff
@@ -1,9 +1,9 @@
 .subheading {
-	margin: 12px 0 30px 0;
+	margin: 12px 0 24px 0;
 	color: var(--token-color-foreground-primary);
 
 	@media (--dev-dot-tablet-up) {
-		margin-bottom: 38px;
+		margin-bottom: 32px;
```
Comment (pbortnick) [inline src/views/profile/bookmarks-view/bookmarks-view.module.css:6]: Subtask: [Adjust spacing](https://app.asana.com/0/0/1203075873125157/f)

File: src/views/profile/bookmarks-view/components/sidebar/index.tsx [hunk 5]
```diff
@@ -24,6 +25,7 @@ export function ProfileBookmarksSidebar() {
 								title: 'Bookmarks',
 								fullPath: '/profile/bookmarks',
 								isActive: true,
+								trailingIcon: <IconBookmark16 />,
```
Comment (pbortnick) [inline src/views/profile/bookmarks-view/components/sidebar/index.tsx:28]: Subtask: [Add bookmark icon next to sidebar menu item](https://app.asana.com/0/0/1203075873125155/f)

File: src/views/profile/bookmarks-view/index.tsx [hunk 6]
```diff
@@ -113,7 +113,12 @@ const ProfileBookmarksViewContent = () => {
 					heading="You have no saved bookmarks."
 					subheading="You can select the bookmark icon on any tutorial card to save it for future reference."
 					callToAction={
-						<ButtonLink href="/tutorials/library" text="Tutorial library" />
+						<ButtonLink
+							href="/tutorials/library"
+							text="Tutorial library"
+							color="secondary"
+							size="small"
+						/>
```
Comment (pbortnick) [inline src/views/profile/bookmarks-view/index.tsx:121]: Subtask: [Make empty state cta button 'secondary' styles](https://app.asana.com/0/0/1203075873125156/f)

File: package.json [hunk 7]
```diff
@@ -92,7 +92,7 @@
 		"@hashicorp/localstorage-polyfill": "^1.0.14",
 		"@hashicorp/mktg-global-styles": "^4.2.3",
 		"@hashicorp/mktg-logos": "^1.3.1",
-		"@hashicorp/platform-analytics": "^0.6.1",
+		"@hashicorp/platform-analytics": "^0.8.0-canary-2022830143359",
```
Comment (pbortnick) [inline package.json:95]: This fixes [Profile dropdown account settings open new window](https://app.asana.com/0/0/1203075873125168/f)

See [this PR](https://github.com/hashicorp/web-platform-packages/pull/96) for more details.

File: src/components/mobile-menu-container/components/mobile-user-disclosure/index.tsx [hunk 8]
```diff
@@ -31,7 +35,7 @@ const renderItem = (
 	if (href) {
 		content = (
 			<Link href={href}>
-				<a className={s.link}>
+				<a className={s.link} rel={rel} target={target}>
```
Comment (pbortnick) [inline src/components/mobile-menu-container/components/mobile-user-disclosure/index.tsx:38]: Fixes the Account Settings link in the mobile menu

**Review by braydenlove — APPROVED** [review]
Looks great thanks so much @pbortnick. FYI the feedback was collected [here](https://docs.google.com/document/d/1UC-jyQ4X_DfigL0GBew2BD-c-_8WPqjsPYZxfA-VBT8/edit#) for those who are curious

File: package.json [hunk 9]
```diff
@@ -92,7 +92,7 @@
 		"@hashicorp/localstorage-polyfill": "^1.0.14",
 		"@hashicorp/mktg-global-styles": "^4.2.3",
 		"@hashicorp/mktg-logos": "^1.3.1",
-		"@hashicorp/platform-analytics": "^0.6.1",
+		"@hashicorp/platform-analytics": "^0.8.0",
```
[the remark this answers is not in the document]
Comment (kendallstrautman) [reply package.json:95]: https://github.com/hashicorp/web-platform-packages/pull/96

**Review by kellyschummer — APPROVED** [review]
Updates are looking over here. Thank you!

File: src/components/bookmark-button/toast/make-bookmark-toast.tsx (hunk 1 above)
Comment (kendallstrautman) [reply src/components/bookmark-button/toast/make-bookmark-toast.tsx:50]: Looks like the toasts still aren't dismissing as expected 🤔 for me they persist indefinitely. 

I wonder if the package was updated.[ Looking at the docs here](https://react-hot-toast.com/docs/toast) it seems like there isn't an `autoDismiss` property anymore. Maybe we need to use `duration`?

Comment (kendallstrautman) [reply src/components/bookmark-button/toast/make-bookmark-toast.tsx:50]: Also could we make the same update on the user progress toast also? Whatever it ends up being — they should be the same duration.

File: src/lib/generate-top-level-sub-nav-items.ts (hunk 3 above)
Comment (kendallstrautman) [reply src/lib/generate-top-level-sub-nav-items.ts:46]: Did some testing, and its looking good to me! I looked at docs, tutorials views at various sizes.

File: src/components/bookmark-button/toast/make-bookmark-toast.tsx (hunk 1 above)
Comment (pbortnick) [reply src/components/bookmark-button/toast/make-bookmark-toast.tsx:50]: So our local toast component should be using the `autoDismiss` prop to set the `duration`. Not sure why it isn't setting but I'll continue looking into it!

https://github.com/hashicorp/dev-portal/blob/main/src/components/toast/index.tsx#L30

File: src/lib/generate-top-level-sub-nav-items.ts (hunk 3 above)
Comment (pbortnick) [reply src/lib/generate-top-level-sub-nav-items.ts:46]: @kendallstrautman thank you!

**Review by kendallstrautman — APPROVED** [review]
👍

**Pull request merged**